src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php

```diff
@@ -98,7 +98,13 @@ public function extractSubmittedData(FormInterface $form): array
             while (null !== $cause) {
                 if ($cause instanceof ConstraintViolationInterface) {
                     $errorData['trace'][] = $cause;
-                    $cause = method_exists($cause, 'getCause') ? $cause->getCause() : null;
+
+                    if (method_exists($cause, 'getCause')) {
+                        $cause = $cause->getCause();
+                    } else {
+                        trigger_deprecation('symfony/validator', '6.2', 'Not implementing the getCause() method in "%s" is deprecated.', $cause::class);
```

Comment:
should not be added here. DebugClassLoader will already warn (once) on class declaration. Warning it would only generate more duplicates.

Reply:
Ah, I wasn't fully sure about this.